### PR TITLE
TATB - 39 [BUG]: Wrap Dialog Data into one config object for more readability

### PR DESCRIPTION
**Pull request by PrabhatJTDeveloper**

### CODE REVIEW

File: src/app/authorization/login/login.component.ts [hunk 1]
```diff
@@ -45,9 +45,8 @@ export class LoginComponent implements OnInit, OnDestroy {
   ngOnInit(): void {
     this.createForm();
     if (localStorage.getItem('registerUser')) {
-      this.userData = JSON.parse(
-        localStorage.getItem('registerUser') as string
-      );
+      let data: any = localStorage.getItem('registerUser');
+      this.userData = JSON.parse(data);
```
Comment (charmijtdev) [inline src/app/authorization/login/login.component.ts]: could you please change this to JSON.parse(localstore.getitem('register user')as string)

File: src/app/component/header/header.component.html [hunk 2]
```diff
@@ -1,22 +1,62 @@
...
+<nav
+  class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
+  style="padding: 0px"
+>
+  <div class="container-fluid">
+    <a
+      href="#"
+      class="navbar-brand h1"
+      routerLink="/dashboard"
+      style="margin: 0px"
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: why inline css

File: src/app/component/header/header.component.html [hunk 3]
```diff
@@ -1,22 +1,62 @@
...
-          >
-            {{ userName }}
-          </a>
+<nav
+  class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
+  style="padding: 0px"
+>
+  <div class="container-fluid">
+    <a
+      href="#"
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: no need for href tag

File: src/app/component/header/header.component.html [hunk 4]
```diff
@@ -1,22 +1,62 @@
...
+      class="navbar-brand h1"
+      routerLink="/dashboard"
+      style="margin: 0px"
+    >
+      <img
+        class="d-inline-block"
+        src="assets/img/logo1.png"
+        width="35"
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: why width height on same please change to external css

File: src/app/component/header/header.component.html [hunk 5]
```diff
@@ -1,22 +1,62 @@
...
+        height="35"
+      />
+      <span>APPLICATION</span></a
+    >
+
+    <div class="navigation">
+      <div class="container-fluid">
+        <div class="collapse navbar-collapse">
+          <img src="assets/img/logo1.png" alt="" class="img-responsive" />
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: please provide alt here

File: src/app/component/header/header.component.html [hunk 6]
```diff
@@ -1,22 +1,62 @@
...
+          <div ngbDropdown class="d-inline-block">
+            <a
+              type="button"
+              class="nav-link dropdown-toggle"
+              id="dropdownBasic1"
+              style="color: rgb(116, 116, 116)"
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: why inline css please change this

File: src/app/component/header/header.component.html [hunk 7]
```diff
@@ -1,22 +1,62 @@
...
+              ngbDropdownToggle
+            >
+              {{ userName }}
+            </a>
+            <div
+              ngbDropdownMenu
+              aria-labelledby="dropdownBasic1"
+              style="min-width: auto; border: none"
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: why inline css ?

File: src/app/component/header/header.component.ts [hunk 8]
```diff
@@ -27,12 +31,31 @@ export class HeaderComponent {
     this.userData = JSON.parse(data);
     let userId: any = this.authenticationService.getUser();
     this.userName = this.userData?.find(
-      (data: any) => data?.id == userId
+      (data: any) => data?.id
```
Comment (charmijtdev) [inline src/app/component/header/header.component.ts]: why removed condtiion ? how i can get the specified user ?

File: src/app/component/header/header.component.ts [hunk 9]
```diff
@@ -27,12 +31,31 @@ export class HeaderComponent {
...
     )?.fullName;
   }
+
+  about(){
+    let configData = this.dialogData.aboutModel;
+      this.dialogService
+        .openDialog(configData)
+        .then((value) => {
+          if (value) {
```
Comment (charmijtdev) [inline src/app/component/header/header.component.ts]: wht is use for value here ?

File: src/app/dialog-service/dialog.service.ts [hunk 10]
```diff
@@ -1,39 +1,30 @@
...
   providedIn: 'root',
 })
 export class DialogService {
   
-  constructor(private modalService: NgbModal, private route: Router) {}
-
-  dialogData = { ...dialogData };
+  constructor(
+    private modalService:NgbModal,
+
```
Comment (charmijtdev) [inline src/app/dialog-service/dialog.service.ts]: please remove this empty space and remove comma after NgbModal

File: src/app/dialog-service/modal/modal/modal.component.html [hunk 11]
```diff
@@ -1,4 +1,64 @@
...
+    (click)="activeModal.close(true)"
+  >
+    {{ configData?.yes }}
+  </button>
+</div>
+</div>
+</div>
+</div>
+
+<!-- <div class="container-fluid">
```
Comment (charmijtdev) [inline src/app/dialog-service/modal/modal/modal.component.html]: please remove this commented code

File: src/app/dialog-service/modal/modal/modal.component.html [hunk 12]
```diff
@@ -1,4 +1,64 @@
...
+  </button>
+  <button
+    *ngIf="configData?.yes"
+    type="button"
+    class="btn btn-success yes-btn"
+    (click)="activeModal.close(true)"
+  >
+    {{ configData?.yes }}
+  </button>
+</div>
```
Comment (charmijtdev) [inline src/app/dialog-service/modal/modal/modal.component.html]: do code formatting

**Review by charmijtdev — CHANGES_REQUESTED** [review]

File: src/app/authorization/login/login.component.ts (hunk 1 above)
Comment (PrabhatJTDeveloper) [reply src/app/authorization/login/login.component.ts]: yes, i changed it.

File: src/app/component/header/header.component.html (hunk 2 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.html]: yes, i removed it

File: src/app/component/header/header.component.html (hunk 3 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.html]: ok, i removed it.

File: src/app/component/header/header.component.html (hunk 4 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.html]: yes. i removed it.

File: src/app/component/header/header.component.html (hunk 5 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.html]: ok, i changed it.

File: src/app/component/header/header.component.html (hunk 6 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.html]: ok, i changed it.

File: src/app/component/header/header.component.html (hunk 7 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.html]: i removed it.

File: src/app/component/header/header.component.ts (hunk 8 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.ts]: yes, i added it.

File: src/app/component/header/header.component.ts (hunk 9 above)
Comment (PrabhatJTDeveloper) [reply src/app/component/header/header.component.ts]: there is no use for it so i removed it.

File: src/app/dialog-service/dialog.service.ts (hunk 10 above)
Comment (PrabhatJTDeveloper) [reply src/app/dialog-service/dialog.service.ts]: yes, i removed it.

File: src/app/dialog-service/modal/modal/modal.component.html (hunk 11 above)
Comment (PrabhatJTDeveloper) [reply src/app/dialog-service/modal/modal/modal.component.html]: yes, i removed it.

File: src/app/dialog-service/modal/modal/modal.component.html (hunk 12 above)
Comment (PrabhatJTDeveloper) [reply src/app/dialog-service/modal/modal/modal.component.html]: yes, i have done this.

File: src/app/authorization/login/login.component.ts [hunk 13]
```diff
@@ -6,24 +6,21 @@ import { AuthenticationService } from 'src/app/service/authentication.service';
 import { ToastService } from 'src/app/toast.service';
 import dialogData from 'src/assets/json/dialogData.json';
 
-
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
```
Comment (charmijtdev) [inline src/app/authorization/login/login.component.ts]: please remove comma

File: src/app/component/header/header.component.html [hunk 14]
```diff
@@ -1,22 +1,41 @@
...
-            type="button"
-            class="nav-link dropdown-toggle"
-            id="dropdownBasic1"
-            ngbDropdownToggle
-          >
-            {{ userName }}
-          </a>
+<nav
+  class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
+  style="padding: 0px"
```
Comment (charmijtdev) [inline src/app/component/header/header.component.html]: please remove inline styling

File: src/app/component/header/header.component.ts [hunk 15]
```diff
@@ -1,25 +1,27 @@
...
-import { AuthenticationService } from '../../service/authentication.service';
+import { DialogService } from 'src/app/dialog-service/dialog.service';
+import dialogData from 'src/assets/json/dialogData.json';
+import { AuthenticationService } from 'src/app/service/authentication.service';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss']
+  styleUrls: ['./header.component.scss'],
```
Comment (charmijtdev) [inline src/app/component/header/header.component.ts]: please remove comma

File: src/app/component/header/header.component.ts [hunk 16]
```diff
@@ -30,9 +32,20 @@ export class HeaderComponent {
       (data: any) => data?.id == userId
     )?.fullName;
   }
+
+  about() {
```
Comment (charmijtdev) [inline src/app/component/header/header.component.ts]: change function name to openAboutDialog()

**Review by charmijtdev — CHANGES_REQUESTED** [review]

**Review by charmijtdev — APPROVED** [review]

**Pull request merged**